release/nightly_tests/setup_cluster_compute_config_updater.py

```diff
@@ -0,0 +1,147 @@
+        ).config.to_dict()
+        logging.info(
+            f"Fetched compute config {self.compute_config_dict} "
+            f"for cluster {self.cluster}"
+        )
+
+    def update(self, field_path: List[Union[str, int]], field_value: Union[str, int]):
+        current = self.compute_config_dict
+        for field in field_path[:-1]:
+            current = current[field]
+        current[field_path[-1]] = field_value
+
+        new_compute_config = anyscale.compute_config.models.ComputeConfig.from_dict(
+            self.compute_config_dict
+        )
+        new_compute_config_name = anyscale.compute_config.create(
+            new_compute_config, name=None
+        )
+        new_compute_config_id = anyscale.compute_config.get(
+            name=new_compute_config_name
+        ).id
+
+        response = requests.put(
+            f"https://console.anyscale-staging.com/api/v2/sessions/{self.cluster_id}/"
+            "cluster_config_with_session_idle_timeout",
+            params={
+                "build_id": self.cluster.result.cluster_environment_build_id,
+                "compute_template_id": new_compute_config_id,
+            },
+            headers={"Authorization": f"Bearer {os.environ['ANYSCALE_CLI_TOKEN']}"},
+        )
+
+        logging.info(
+            f"Update compute config to {self.compute_config_dict}, "
+            f"got response {response}"
+        )
+        response.raise_for_status()
+
+
+@ray.remote(num_cpus=0)
+class ClusterComputeConfigUpdater:
+    def __init__(self, updates: List[ClusterComputeConfigUpdate]):
+        self.cluster_compute_config = ClusterComputeConfig()
+        self.updates = updates
+        self.start_time = None
+
+    async def run(self):
+        logging.info("Start to run")
+        self.start_time = time.monotonic()
+        while self.updates:
+            delay = (
+                self.start_time + self.updates[0].update_time_seconds_since_run
+            ) - time.monotonic()
+            if delay > 0:
+                logging.info(f"Sleep for {delay} seconds")
+                await asyncio.sleep(delay)
+            self.cluster_compute_config.update(
+                self.updates[0].field_path, self.updates[0].field_value
+            )
+            self.updates.pop(0)
+
+    async def wait_until_run(self):
+        while not self.start_time:
+            await asyncio.sleep(0.1)
+
+
+if __name__ == "__main__":
+    ray.init(logging_config=ray.LoggingConfig(encoding="TEXT"))
+    arg_parser = argparse.ArgumentParser()
+
+    arg_parser.add_argument(
+        "--updates",
+        type=str,
+        default="",
+        help=(
+            "A comma separated list of updates, each update has format: "
+            "field_path:field_value:update_time_seconds_since_run."
+            "field_path is a dot separated list of fields from root "
+            "to the target field to be updated, e.g. worker_nodes.0.max_nodes."
+        ),
```

Comment:
Nit: Rather than expecting a comma-separate listed and splitting the result, I think we can do `nargs="+"`